client/src/Visit.tsx

```diff
@@ -55,45 +66,50 @@ export const Visit = (): JSX.Element => {
     setMeetingLinkLocator(meetingLinkLocator);
   };
 
+  const [config, setConfig] = useState<AppConfigModel | undefined>(undefined);
+  const [error, setError] = useState<any | undefined>(undefined);
+  const [meetingLinkLocator, setMeetingLinkLocator] = useState<TeamsMeetingLinkLocator | undefined>(
+    _getMeetingLinkLocator(window.location.search) // case of direct link to visit with teams meeting link in URL
+  );
+  const [roomsLocator] = useState<RoomCallLocator | undefined>(
+    _getRoomLocator(window.location.search) // case of direct link to visit with rooms meeting link in URL
+  );
+  const [participantId] = useState<string | undefined>(_getParticipantId(window.location.search)); // case of direct link to visit with rooms meeting link in URL
+
   useEffect(() => {
     const fetchData = async (): Promise<void> => {
       try {
         const config = await fetchConfig();
-        const token = await fetchToken();
         setConfig(config);
-        setToken(token);
       } catch (error) {
         console.error(error);
         setError(error);
       }
     };
-
     fetchData();
   }, []);
 
   if (error) {
     return <GenericError statusCode={error.statusCode} />;
   }
 
-  if (!config || !token) {
-    // config and token not ready yet - show spinning/loading animation
+  if (!config) {
+    // config is not ready yet - show spinning/loading animation
     return <Spinner styles={fullSizeStyles} />;
   }
 
   if (!meetingLinkLocator || !meetingLinkLocator.meetingLink) {
-    // If we have config and token but don't have a meeting link,
-    // show a separate screen with "enter meeting link" textbox
-    return (
-      <ThemeProvider theme={config.theme} style={{ height: '100%' }}>
-        <JoinTeamsMeeting config={config} onJoinMeeting={(link) => _onJoinMeeting(link)} />
-      </ThemeProvider>
-    );
+    if (!roomsLocator || !participantId) {
```

Comment:
These conditions are reused in this file. For better readability, can we refactor this to:

```
const isTeamsMeeting = meetingLinkLocator && meetingLinkLocator.meetingLink;
const isRoomsMeeting = roomsLocator && participantId;

if (!isTeamsMeeting && !isRoomsMeeting) {
    render JoinTeamsMeeting
}
```